packages/snaps-execution-environments/src/webworker/pool/WebWorkerPool.test.browser.ts

```diff
@@ -0,0 +1,188 @@
+import { MockPostMessageStream, spy } from '@metamask/snaps-utils/test-utils';
+import { assert } from '@metamask/utils';
+
+import { WebWorkerPool } from './WebWorkerPool';
+
+const MOCK_JOB_ID = 'job-id';
+const WORKER_URL = 'http://localhost:4568/worker/executor/';
+
+/**
+ * Write a message to the stream, wrapped with the job ID and frame URL.
+ *
+ * @param stream - The stream to write to.
+ * @param message - The message to write.
+ * @param jobId - The job ID.
+ */
+function writeMessage(
```

Comment:
Could we move these to a shared file or is there not enough overlap between implementations?